clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -402,6 +412,30 @@ public synchronized void maybeAddPartition(TopicPartition topicPartition) {
         }
     }
 
+    public synchronized void maybeHandlePartitionAdded(TopicPartition topicPartition) {
```

Comment:
For my understanding, this was the previous handling for the add partition call, but since add partition is now part of the produce request, we have a separate block?

I wonder if we should change some of these error messages to reflect that the record was not written either?

Reply:
Refactored this part, and removed the previous handling.
This method is called as a part of the produce callback and if there is any error in the produceResponse, it will not be called.